### PR TITLE
Fixing bugs in MultiWorkUnit

### PR DESCRIPTION
**Pull request by sahilTakiar**
Duplicate of https://github.com/linkedin/gobblin/pull/208

@liyinan926 I addressed your comments.

### CODE REVIEW

File: gobblin-runtime/src/main/java/gobblin/runtime/mapreduce/MRJobLauncher.java [hunk 1]
```diff
@@ -401,15 +401,18 @@ private Path prepareJobInput(Path jobInputPath, List<WorkUnit> workUnits) throws
       int multiTaskIdSequence = 0;
       // Serialize each work unit into a file named after the task ID
       for (WorkUnit workUnit : workUnits) {
+
+        Path workUnitFileExt;
```
Comment (liyinan926) [inline gobblin-runtime/src/main/java/gobblin/runtime/mapreduce/MRJobLauncher.java]: You don't need even to make a `Path`, a `String` is sufficient. Also please rename it to `workUnitFileName`.

**Comment by liyinan926** [comment]
LGTM other than the only comment.

**Comment by sahilTakiar** [comment]
@ibuenros any comments?

**Comment by ibuenros** [comment]
It looks good. Is there any use case where we would want to set a property in the MultiWorkUnit, but not for it to spread to the WorkUnits? If so, we could add a separate method to allow for that, but I can't  think of any such use case.

**Comment by sahilTakiar** [comment]
I can't think of a use case either, but it may be nice to have, I can add such a method.

**Comment by sahilTakiar** [comment]
Updated

**Comment by ibuenros** [comment]
LGTM

**Pull request merged**